http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriberSpec.scala

```diff
@@ -1408,6 +1405,96 @@ class LeoPubsubMessageSubscriberSpec
+    var deleteCalled = false
+
+    val mockKubernetesService = new MockKubernetesService(PodStatus.Succeeded) {
+      override def createServiceAccount(
+        clusterId: GKEModels.KubernetesClusterId,
+        serviceAccount: KubernetesModels.KubernetesServiceAccount,
+        namespaceName: KubernetesModels.KubernetesNamespace
+      )(implicit ev: ApplicativeAsk[IO, TraceId]): IO[Unit] =
+        IO.raiseError(new Exception("this is an intentional test exception"))
+
+      override def deleteNamespace(
+        clusterId: GKEModels.KubernetesClusterId,
+        namespace: KubernetesModels.KubernetesNamespace
+      )(implicit ev: ApplicativeAsk[IO, TraceId]): IO[Unit] =
+        IO {
+          deleteCalled = true
+        }
+    }
+
+    val gkeInterp =
+      new GKEInterpreter[IO](Config.gkeInterpConfig,
+                             vpcInterp,
+                             MockGKEService,
+                             mockKubernetesService,
+                             MockHelm,
+                             MockGalaxyDAO,
+                             credentials,
+                             blocker)
+
+    val assertions = for {
+      clusterOpt <- kubernetesClusterQuery.getMinimalClusterById(savedCluster1.id).transaction
+      getCluster = clusterOpt.get
+      getAppOpt <- KubernetesServiceDbQueries
+        .getFullAppByName(savedCluster1.googleProject, savedApp1.id)
+        .transaction
+      getApp = getAppOpt.get
+      getDiskOpt <- persistentDiskQuery.getById(savedApp1.appResources.disk.get.id).transaction
+      getDisk = getDiskOpt.get
+    } yield {
+      getCluster.status shouldBe KubernetesClusterStatus.Running
+      //only the default should be left, the other has been deleted
+      getCluster.nodepools.size shouldBe 1
+      getCluster.nodepools.filter(_.isDefault).head.status shouldBe NodepoolStatus.Running
+      getApp.app.errors.size shouldBe 1
+      getApp.app.status shouldBe AppStatus.Error
+      getApp.nodepool.status shouldBe NodepoolStatus.Deleted
+      getDisk.status shouldBe DiskStatus.Deleted
+      deleteCalled shouldBe true
+    }
+
+    val res = for {
+      tr <- traceId.ask
+      dummyNodepool = savedCluster1.nodepools.filter(_.isDefault).head
+      msg = CreateAppMessage(
+        Some(CreateCluster(savedCluster1.id, dummyNodepool.id)),
+        savedApp1.id,
+        savedApp1.appName,
+        Some(savedNodepool1.id),
+        savedCluster1.googleProject,
+        Some(disk.id),
+        Map.empty,
+        Some(tr)
+      )
+      queue <- InspectableQueue.bounded[IO, Task[IO]](10)
+      leoSubscriber = makeLeoSubscriber(asyncTaskQueue = queue, gkeInterp = gkeInterp)
+      asyncTaskProcessor = AsyncTaskProcessor(AsyncTaskProcessor.Config(10, 10), queue)
+      _ <- leoSubscriber.handleCreateAppMessage(msg)
+      _ <- withInfiniteStream(asyncTaskProcessor.process, assertions, maxRetry = 50)
+    } yield ()
+
+    res.unsafeRunSync()
+  }
+
```

Comment:
Would it be good to test the scenario where the disk pre-existed and wouldn't be cleaned up?